integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/clone/CloneFunctionalContextTestCase.java

```diff
@@ -55,21 +60,42 @@ public void testSequence() throws Exception {
         //Added to ensure that carbon log is updated with required entries
         TimeUnit.SECONDS.sleep(10);
 
-        LogEvent[] getLogsInfo = logViewer.getAllRemoteSystemLogs();
-        boolean assertValue = false;
-        for (LogEvent event : getLogsInfo) {
-            if (event.getMessage().contains("REQUEST PARAM VALUE")) {
-                assertValue = true;
-                break;
-            }
-        }
-        Assert.assertTrue(assertValue, "Synapse functional context not cloned.");
+        Awaitility.await()
+                  .pollInterval(500, TimeUnit.MILLISECONDS)
+                  .atMost(300, TimeUnit.SECONDS)
+                  .until(isMessageRecived());
     }
 
     @AfterClass(alwaysRun = true)
     public void close() throws Exception {
         super.cleanup();
     }
-
+    /**
+     * Check if message received from server logs.
+     *
+     * @return
+     */
+    private Callable<Boolean> isMessageRecived() {
+        return new Callable<Boolean>() {
+            @Override
+            public Boolean call() {
+                LogEvent[] getLogsInfo;
+                boolean assertValue = false;
+                try {
+                    getLogsInfo = logViewer.getAllSystemLogs();
```

Comment:
Don't we have a Util method for this? If not, shall we extract this to a util method

Reply:
Fixed